pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/intervalST/Interval.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.routing.segmentpruner.intervalST;
+
+import com.google.common.base.Preconditions;
+import javax.validation.constraints.NotNull;
+
+
+/**
+ * The {@code Interval} class represents an one-dimensional closed interval which contains both ends.
+ */
+public class Interval implements Comparable<Interval> {
+  public final long _min;
+  public final long _max;
+
+  public Interval(long min, long max) {
+    Preconditions.checkState(min <= max, "invalid interval [{}, {}]", min, max);
+    _min = min;
+    _max = max;
+  }
+
+  public boolean intersects(@NotNull Interval that) {
+    return _max >= that._min && that._max >= _min;
+  }
+
+  @Override
+  public int compareTo(Interval that) {
+    Preconditions.checkNotNull(that, "Compare to invalid interval: null");
+    if (_min < that._min) {
+      return -1;
+    } else if (_min > that._min) {
+      return 1;
+    } else if (_max < that._max) {
+      return -1;
+    } else if (_max > that._max) {
+      return 1;
+    }
+    else return 0;
+  }
+
+  @Override
+  public int hashCode() {
+    return (int)(_min * 17 + _max);
+  }
+
+  @Override
+  public boolean equals(Object that) {
```

Comment:
Is this intellij auto generated `equals()`? If not, let's use the one generated by intellij.

Reply:
No, I'll use the auto generated one instead.